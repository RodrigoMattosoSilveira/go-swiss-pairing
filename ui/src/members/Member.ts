export class Member {
    id: string | undefined;
    first: string = '';
    last: string = '';
    email: string = '';
    password: string = '';
    cell: string = '';
    rating: number = 1200;
    isActive: boolean = false
    imageUrl: string = 'picture_placeholder.jpeg';

    constructor(initializer?: any) {
        if (!initializer) return;
        if (initializer.id) this.id = initializer.id;
        if (initializer.first) this.first = initializer.first;
        if (initializer.last) this.last = initializer.last;
        if (initializer.email) this.email = initializer.email;
        if (initializer.password) this.password = initializer.password;
        if (initializer.cell) this.cell = initializer.cell;
        if (initializer.rating) this.rating = initializer.rating;
        if (initializer.isActive) this.isActive = initializer.isActive;
        if (initializer.imageUrl) this.imageUrl = initializer.imageUrl;
    }
}
