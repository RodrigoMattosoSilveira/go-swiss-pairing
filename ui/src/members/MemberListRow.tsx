import React from 'react';
import { Link } from 'react-router-dom';
import { Member } from './Member';

interface MemberListRowProps {
    member: Member;
    onEdit: (member: Member) => void;
}

function MemberListRow(props: MemberListRowProps) {
    // return <pre>{JSON.stringify(members, null, ' ')}</pre>;
    const { member, onEdit } = props;
    const handleEditClick = (memberBeingEdited: Member) => {
        // console.log(memberBeingEdited);
        onEdit(memberBeingEdited);
    };
    return (
        <>
            <div className="col-sm">
                <button
                    className="small"
                    onClick={() => {
                        handleEditClick(member);
                    }}
                >
                    <span className="icon-edit"></span>
                </button>
            </div>
            <Link to={'/members/' + member.id}> <div className="col-sm">{member.id}</div></Link>
            <div className="col-sm">{member.first}</div>
            <div className="col-sm">{member.last}</div>
            <div className="col-sm">{member.email}</div>
            <div className="col-sm">{member.cell}</div>
            <div className="col-sm">{member.password}</div>
            <div className="col-sm">{member.rating}</div>
            <div className="col-sm">{member.isActive ? "Yes" : "No"}</div>
        </>
    )
}

export default MemberListRow;