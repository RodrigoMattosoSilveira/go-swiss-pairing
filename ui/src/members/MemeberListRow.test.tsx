import { render, screen } from "@testing-library/react";
import React from "react";
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { Member } from "./Member";
import MemberListRow from "./MemberListRow";
import renderer from "react-test-renderer";

describe("Member List Row", () => {
    let member: Member;
    let handleEdit: jest.Mock;

    const setup = () =>
        render(
            <MemoryRouter>
                <MemberListRow member={member} onEdit={handleEdit} />
            </MemoryRouter>
        );
    beforeEach(() => {
        member = new Member({
            "id":"mkq9OLj-Wf",
            "first":"Arianna",
            "last":"Moody",
            "email":"Arianna.Moody@yahoo.com",
            "password":"VH8q4tkg6UX3",
            "cell":"801 277-6891",
            "rating":1026,
            "isActive":true,
            "imageUrl":"/assets/picture_mkq9OLj-Wf.jpeg"},);
        handleEdit = jest.fn();
    });

    it("should initially render", () => {
        setup();
    });
    it("Member List Row member={member} onEdit={handleEdit}", () => {
        // Adding the Router is a hack to make this test work
        const tree = renderer
            .create(
                <MemoryRouter>
                    <MemberListRow member={member} onEdit={handleEdit} />
                </MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
    it('renders project properly', () => {
        setup();
        expect(screen.getByRole('generic', { name: 'member.email' })).toHaveTextContent(member.email);
        // screen.debug(document);
        screen.getByText(/Arianna.Moody@yahoo\.com/i);
        screen.getByText(/1026/i);
    });

    it('handler called when edit clicked', async () => {
        setup();
        // this query works screen.getByText(/edit/i)
        // but using role is better
        const user = userEvent.setup();
        await user.click(screen.getByRole('button', { name: member.id}));
        expect(handleEdit).toBeCalledTimes(1);
        expect(handleEdit).toBeCalledWith(member);
   });
});