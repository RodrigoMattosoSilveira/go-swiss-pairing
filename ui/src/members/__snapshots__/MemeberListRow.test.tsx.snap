// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Row Member List Row member={member} onEdit={handleEdit} 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-1"
  >
    <button
      aria-label="mkq9OLj-Wf"
      className="small"
      onClick={[Function]}
    >
      <span
        className="icon-edit"
      />
    </button>
  </div>
  <a
    aria-label="member.id"
    className="col-sm-1"
    href="/members/mkq9OLj-Wf"
    onClick={[Function]}
  >
    mkq9OLj-Wf
  </a>
  <div
    aria-label="member.first"
    className="col-sm-1"
  >
    Arianna
  </div>
  <div
    aria-label="member.last"
    className="col-sm-1"
  >
    Moody
  </div>
  <div
    aria-label="member.email"
    className="col-sm-2"
  >
    Arianna.Moody@yahoo.com
  </div>
  <div
    aria-label="member.cell"
    className="col-sm-2"
  >
    801 277-6891
  </div>
  <div
    aria-label="member.password"
    className="col-sm-2"
  >
    VH8q4tkg6UX3
  </div>
  <div
    aria-label="member.rating"
    className="col-sm-1"
  >
    1026
  </div>
  <div
    aria-label="member.isActive"
    className="col-sm-1"
  >
    Yes
  </div>
</div>
`;
